backend/src/main/kotlin/fr/gouv/cacem/monitorenv/infrastructure/api/adapters/bff/inputs/missions/MissionEnvActionDataInput.kt

```diff
@@ -0,0 +1,95 @@
+package fr.gouv.cacem.monitorenv.infrastructure.api.adapters.bff.inputs.missions
+
+import fr.gouv.cacem.monitorenv.domain.entities.VehicleTypeEnum
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.ActionTypeEnum
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.EnvActionEntity
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.EnvActionNoteEntity
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.EnvActionSurveillanceEntity
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.ThemeEntity
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.envActionControl.ActionTargetTypeEnum
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.envActionControl.EnvActionControlEntity
+import org.locationtech.jts.geom.Geometry
+import java.time.ZonedDateTime
+import java.util.Optional
+import java.util.UUID
+
+data class MissionEnvActionDataInput(
+    val id: UUID,
+    val actionType: ActionTypeEnum,
+    val actionStartDateTimeUtc: ZonedDateTime,
+    val actionEndDateTimeUtc: ZonedDateTime? = null,
+    val department: String? = null,
+    val facade: String? = null,
+    val geom: Geometry? = null,
+
+    // Common to all action Types
+    val observations: String? = null,
+
+    // EnvActionControl + EnvSurveillance Properties
+    val themes: List<ThemeEntity>? = null,
+
+    // EnvActionControl Properties
+    val actionNumberOfControls: Int? = null,
+    val actionTargetType: ActionTargetTypeEnum? = null,
+    val vehicleType: VehicleTypeEnum? = null,
+    val infractions: List<MissionEnvActionControlInfractionDataInput>? = listOf(),
+
+    // EnvActionSurveillance Properties
+    val coverMissionZone: Boolean? = null,
+
+    // complementary properties
+    val reportingIds: Optional<List<Int>>,
+) {
+    fun checkValidity() {
+        when (actionType) {
+            ActionTypeEnum.CONTROL -> require(this.reportingIds.isPresent && this.reportingIds.get().size < 2) {
+                "ReportingIds must not be empty and maximum 1 id for Controls"
+            }
+            ActionTypeEnum.SURVEILLANCE -> require(this.reportingIds.isPresent) {
+                "ReportingIds must not be empty for Surveillance"
```

Comment:
Il ne doit pas être `null` mais peut être vide, non ? (C'est le terme `empty` : "reportingsIds parameter must be given" ? )